src/workerd/api/streams/standard.c++

```diff
@@ -335,28 +347,35 @@ ReadableImpl<Self>::ReadableImpl(
 template <typename Self>
 void ReadableImpl<Self>::start(jsg::Lock& js, jsg::Ref<Self> self) {
   KJ_ASSERT(!started && algorithms.starting == nullptr);
-  auto& ioContext = IoContext::current();
 
-  auto onSuccess = ioContext.addFunctor(
-      JSG_VISITABLE_LAMBDA((this, self = self.addRef()), (self), (jsg::Lock& js) {
+  auto onSuccess = JSG_VISITABLE_LAMBDA((this, self = self.addRef()), (self), (jsg::Lock& js) {
     algorithms.starting = nullptr;
     started = true;
     pullIfNeeded(js, kj::mv(self));
-  }));
+  });
 
-  auto onFailure = ioContext.addFunctor(
-      JSG_VISITABLE_LAMBDA((this, self = self.addRef()), (self),
-                           (jsg::Lock& js, jsg::Value reason) {
+  auto onFailure = JSG_VISITABLE_LAMBDA((this, self = self.addRef()), (self),
+                                        (jsg::Lock& js, jsg::Value reason) {
     algorithms.starting = nullptr;
     started = true;
     doError(js, kj::mv(reason));
-  }));
+  });
+
+  if (IoContext::hasCurrent()) {
+    auto& ioContext = IoContext::current();
+    algorithms.starting = maybeRunAlgorithm(js,
+                                            algorithms.start,
+                                            ioContext.addFunctor(kj::mv(onSuccess)),
+                                            ioContext.addFunctor(kj::mv(onFailure)),
+                                            kj::mv(self));
+  } else {
+    algorithms.starting = maybeRunAlgorithm(js,
+                                            algorithms.start,
+                                            kj::mv(onSuccess),
+                                            kj::mv(onFailure),
+                                            kj::mv(self));
+  }
```

Comment:
Does `maybeRunAlgorithm()` now contain the same logic, so this branch isn't necessary?